File: frontend/src/lib/api/analytics.ts
import { apiClient } from './client';
import type {
  DashboardData,
  Report,
  ReportData,
  SystemMetrics,
} from '@/types/api';

// Analytics API
export const analyticsApi = {
  getDashboard: async (): Promise<DashboardData> => {
    const response = await apiClient.get('/api/v1/analytics/dashboard');
    return response.data;
  },

  listReports: async (): Promise<{ reports: Report[] }> => {
    const response = await apiClient.get('/api/v1/analytics/reports');
    return response.data;
  },

  getReport: async (reportId: string): Promise<ReportData> => {
    const response = await apiClient.get(`/api/v1/analytics/reports/${reportId}`);
    return response.data;
  },

  getMetrics: async (period?: string): Promise<SystemMetrics> => {
    const response = await apiClient.get('/api/v1/analytics/metrics', {
      params: { period },
    });
    return response.data;
  },
};

File: backend/api/v1/analytics.py
"""
Analytics Endpoints

Provides:
- Dashboard data
- Reports
- KPI tracking
"""

from datetime import datetime

from fastapi import APIRouter

from api.schemas.analytics import KPI, DashboardData

router = APIRouter()

@router.get("/dashboard", summary="Get dashboard data")
async def get_dashboard():
    """Get main dashboard data."""
    return DashboardData(
        kpis=[
            KPI(name="Total Revenue", value=2450000, unit="USD", change=12.5, trend="up"),
            KPI(name="Active Customers", value=15420, unit="count", change=5.2, trend="up"),
            KPI(name="Churn Rate", value=2.3, unit="%", change=-0.5, trend="down"),
            KPI(name="Model Accuracy", value=94.5, unit="%", change=1.2, trend="up"),
            KPI(name="Predictions Made", value=125000, unit="count", change=25.0, trend="up"),
            KPI(name="Cost Savings", value=350000, unit="USD", change=18.0, trend="up")
        ],
        charts={
            "revenue_trend": {"labels": ["Jan", "Feb", "Mar", "Apr", "May"], "values": [180000, 195000, 210000, 225000, 245000]},
            "predictions_by_model": {"anomaly": 45000, "churn": 35000, "forecast": 45000},
            "agent_utilization": {"financial": 85, "operations": 72, "customer": 68, "security": 90}
        },
        last_updated=datetime.now()
    )


@router.get("/reports", summary="List reports")
async def list_reports():
    """List available reports."""
    return {
        "reports": [
            {"id": "rep_001", "name": "Monthly Executive Summary", "type": "executive", "frequency": "monthly"},
            {"id": "rep_002", "name": "ML Model Performance", "type": "technical", "frequency": "weekly"},
            {"id": "rep_003", "name": "Customer Analytics", "type": "business", "frequency": "daily"}
        ]
    }


@router.get("/reports/{report_id}", summary="Get report")
async def get_report(report_id: str):
    """Generate and return a report."""
    return {
        "id": report_id,
        "name": "Sample Report",
        "generated_at": datetime.now(),
        "data": {
            "summary": "Platform performance summary",
            "metrics": {"accuracy": 0.94, "uptime": 99.9, "cost_savings": 350000}
        }
    }


@router.get("/metrics", summary="Get system metrics")
async def get_metrics(period: str = "24h"):
    """Get system performance metrics."""
    return {
        "period": period,
        "api_requests": {"total": 125000, "success_rate": 99.8},
        "predictions": {"total": 45000, "avg_latency_ms": 12.5},
        "pipelines": {"active": 8, "records_processed": 2500000},
        "decisions": {"total": 156, "autonomous": 142, "manual": 14}
    }
